### ISSUE TITLE
router-bridge: Use snapshotting to speed up instantiation of a v8 runtime.

### ISSUE THREAD

**Issue by garypen** [apollographql/federation#1299]
We already have an issue to re-use v8 runtimes where possible (see: #1266). This suggestion is complementary and can be either performed in combination with #1266 or independently. The idea is to perform "common" initialisation of the runtime just once and then save a snapshot of that runtime in memory and use in future runtime instantiation.

The belief is that initialisation from snapshot is faster than performing initialisation. I've done some limited testing and found that query performance improves by between 30% and 80%. This is some way from working right now, but merits a bit of investigation.

See draft PR (#1292) for more details.

**Comment by garypen** [comment]
## Router-Bridge and Deno-Core Relationship

As part of investigating use of V8 snapshotting to improve router performance
I fell into small nest of issues which needed to be resolved between:
 - router
 - router-bridge
 - deno-core

### Problem Statement

Snapshot experimentation worked fine on a build machine, but as soon as a
router was run in a non-build environment it would fail with an error
something like:

```
cargo:rerun-if-changed=/Users/garypen/.asdf/installs/rust/1.57.0/registry/src/github.com-1ecc6299db9ec823/deno_core-0.110.0/00_primordials.js
thread 'tokio-runtime-worker' panicked at 'called `Result::unwrap()` on an `Err` value: No such file or directory (os error 2)', /Users/garypen/.asdf/installs/rust/1.57.0/registry/src/github.com-1ecc6299db9ec823/deno_core-0.110.0/runtime.rs:402:38
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'tokio-runtime-worker' panicked at 'NaiveIntrospection instantiation panicked: JoinError::Panic(...)', apollo-router/src/apollo_router.rs:45:18
2021-12-13T16:31:40.037003Z ERROR router: Failed to start server.
Error: Failed to start server.
```

It transpires that no-one has seen this issue in the router because the router
is pinned to an old version of the router-bridge, which used deno-core 0.86.0.
New versions of deno-core (e.g. HEAD is at 0.110.0) are not intended to
support non-snapshot invocations of V8 runtimes. For more (although only
partial) details see [deno core issue](https://github.com/denoland/deno/issues/13067). I tripped over it because my modifications to router-bridge were based
on router-bridge HEAD...

### Resolution

The issue referenced above indicates that due to the change to how deno-core
works, implemented sometime after release 0.86.0, we need to use snapshots
to initialise our V8 VM.

The cleanest approach is to generate a snapshot in build (as part of build.rs),
then include the generated snapshot into the compiled router-bridge.

I've prototyped this work (see PR: #1292) and the approach is working fine.

### Next Steps

Performance testing is required in validate that snapshot based V8 is no
slower than the current approach. Once the changes to router-bridge are
merged, we'd need to update the router to reference the new router-bridge
code.

**Comment by garypen** [comment]
resolved by a PR in federation to resolve the underlying deno-core changes and a PR in the router to start using this new router-bridge:
PR: #1292 
PR: https://github.com/apollographql/router/pull/269
